lib/actions/ExtensionAction.js

```diff
@@ -317,7 +320,12 @@ class ExtensionAction {
       if (!extensionInfo) {
         return logger.warn(`There is no extension in folder: './extensions/${extensionFolderName}'. Skipping... Please make sure that you only pass the folder name of the extension as argument.`)
       }
-      this.settings.attachExtension(extensionFolderName, extensionInfo, force)
+      this.settings.attachExtension(extensionFolderName, {id: extensionInfo.id, trusted: extensionInfo.trusted}, force)
+
+      // Create components.json if needed
+      if (extensionInfo.components && extensionInfo.components.length > 0) {
+        utils.generateComponentsJson(this.settings.getApplicationFolder(), extensionInfo.id, extensionInfo.components)
```

Comment:
I may have got some kind of missing link here - utils.generateComponents() expects one parameter "appSettings", supposedly an AppSettings object. This won't work, will it?

I also think we're missing a test case here that makes sure the method is called/not called under given conditions.

Reply:
missed that one ... done